pkg/hierarchy/wired_clusterqueue.go

```diff
@@ -0,0 +1,19 @@
+package hierarchy
+
+type WiredClusterQueue[CQ, Cohort nodeBase] struct {
```

Comment:
What does "wired" mean?

Add a comment or find a better word.

Reply:
TBH, I would just name this `ClusterQueue`.

The differentiator from the other structs in the project comes from the package name.